### PR TITLE
Fixes for Modular Converter on Windows

### PR DESCRIPTION
**Pull request by hlky**
# What does this PR do?

This PR fixes some issues with Modular Converter on Windows.

1. Relative path regex. `rf"(src{os.sep}transformers{os.sep}.*|examples{os.sep}.*)"` results in `(src\transformers\.*|examples\.*)`, due to `\t` and `\.` this regex fails, we replace `\` on `os.path.abspath` instead, and use `(src/transformers/.*|examples/.*)` as the regex.
~~2. Relative path in auto generated message. On Windows this generates as `This file was automatically generated from src\transformers\models\florence2\modular_florence2.py.` This could potentially cause issues elsewhere, and it's better to be standardized, so when `os.sep` == `\` we replace `os.sep` with `/`.~~
3. `open` encoding. On Windows the default encoding is `cp1252` which doesn't work with `🚨`, so we use `encoding=utf-8` instead for all `open`. This particular fix is in effect in many other places in the codebase.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker

### CODE REVIEW

**Comment by hlky** [comment]
Please note the test failure here is unrelated, looks like `modeling_glm` needs to be overwritten with modular generated code.

File: utils/modular_model_converter.py [hunk 1]
```diff
@@ -1142,9 +1142,12 @@ def convert_modular_file(modular_file, old_model_name=None, new_model_name=None,
         for file, node in cst_transformers.files.items():
             if node != {}:
                 # Get relative path starting from src/transformers/
+                sep = os.sep * 2 if os.sep == "\\" else os.sep
                 relative_path = re.search(
-                    rf"(src{os.sep}transformers{os.sep}.*|examples{os.sep}.*)", os.path.abspath(modular_file)
+                    rf"(src{sep}transformers{sep}.*|examples{sep}.*)", os.path.abspath(modular_file)
```
Comment (ArthurZucker) [inline utils/modular_model_converter.py:1147]: I think we'd better use `os.path.join` no? This would directly support correct separation?

**Review by ArthurZucker — APPROVED** [review]
LGTM maybe using os.path will simplify the separation! 🤗

File: utils/modular_model_converter.py (hunk 1 above)
Comment (hlky) [reply utils/modular_model_converter.py:1147]: It's the same separation as `os.sep` with `os.path.join`. This is a python/regex issue due to the combination of raw and formatted.

An alternate solution would be replacing `\\` on `os.path.abspath`, then regex can just use `/`:
```python
relative_path = re.search(
    r"(src/transformers/.*|examples/.*)", os.path.abspath(modular_file).replace("\\", "/")
).group(1)
```

Comment (hlky) [reply utils/modular_model_converter.py]: Done, thanks!

**Comment by ArthurZucker** [comment]
Thanks for your contribution!

**Pull request merged**